File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
    
        String oriStr = Integer.toBinaryString(n);
        int oriCnt = cntOne(n);
        
        int target = n+1;
        
        while(cntOne(target)!=oriCnt){
            target++;
        }
        
        return target;
    }
    
    public int cntOne(int n){
        int cnt = 0;
        String bitStr = Integer.toBinaryString(n);
        for(int i=0;i<bitStr.length();i++){
            if(bitStr.charAt(i)=='1'){
                cnt++;
            }   
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    curr = n
    find = False
    bistr = bin(curr)
    ori_cnt = 0
    
    for i in bistr:
            if i == "1":
                ori_cnt = ori_cnt + 1
    
    while find == False:
        curr = curr + 1
        bistr = bin(curr)
        cnt = 0
        
        for i in bistr:
            if i == "1":
                cnt = cnt + 1
        
        if ori_cnt == cnt :
            find = True
        
    answer = curr
    return answer